include/boost/url/pct_encoding.hpp

```diff
@@ -59,30 +59,44 @@ struct pct_decode_opts
     for the character set, setting the error if
     the string is invalid.
 
+    @par Example
+    @code
+    error_code ec;
+    auto needed = validate_pct_encoding("Some%20text", ec, pchars);
+
+    assert(!ec);
+    assert(needed == 9);
+    @endcode
+
+    @par Example
+    @code
+    error_code ec;
+    auto needed = validate_pct_encoding("Some%2text", ec, pchars);
+
+    assert(ec);
+    @endcode
+
```

Comment:
In implementation I would use `auto` but in example code I prefer to spell out the variable. And I also use additional spaces inside parenthesis because it renders better in docs. When there are multiple examples I provide a short descriptive sentence. How about:
```
@par Example 1
This validates and calculates the decoded length of a valid percent-encoded string.
@code
error_code ec;
std::size_t decoded_size = validate_pct_encoding( "Program%20Files", ec, pchars );
assert( ! ec.failed() );
assert( decoded_size == 13 );
@endcode

@par Example 2
This shows how validation can fail using an error code:
@code
error_code ec;
std::size_t decoded_size = validate_pct_encoding( "bad%2escape", ec, pchars );
assert( ec.failed() );
@endcode
```